src/app/core/utils/navigation-listener.service.spec.ts

```diff
@@ -0,0 +1,62 @@
+import { NavigationListenerService } from './navigation-listener.service';
+import { AlertService, NavigationNotifierService, NavigationOrigin } from '@hmcts/ccd-case-ui-toolkit';
+import { Router } from '@angular/router';
+import createSpyObj = jasmine.createSpyObj;
+
+describe('NavigationListenerService', () => {
+
+  let navigationListenerService: NavigationListenerService;
+  let alertService: AlertService;
+  let navigationNotifier: NavigationNotifierService;
+  let mockRouter: any;
+
+  beforeEach(() => {
+    alertService = jasmine.createSpyObj<AlertService>('alertService', ['setPreserveAlerts', 'success']);
+    mockRouter = createSpyObj<Router>('router', ['navigate']);
+    mockRouter.navigate.and.returnValue(Promise.resolve(true));
+    navigationNotifier = new NavigationNotifierService();
+    navigationListenerService = new NavigationListenerService(alertService, navigationNotifier, mockRouter);
+  });
+
+  it('test DRAFT_DELETED navigation subscription', () => {
+    let DRAFT_DELETED = NavigationOrigin.DRAFT_DELETED;
+    navigationNotifier.announceNavigation({ action: DRAFT_DELETED });
+    navigationListenerService.init();
+    expect(mockRouter.navigate).toHaveBeenCalledWith(['list/case']);
+  });
+
+  it('test ERROR_DELETING_DRAFT navigation subscription', () => {
```

Comment:
Need to also handler error:
```
  private handleError(error: HttpError, trigger: CaseViewTrigger) {
    if (error.status !== 401 && error.status !== 403) {
      this.error = error;
      console.log('error during triggering event:', trigger.id);
      console.log(error);
      this.callbackErrorsSubject.next(this.error);
    }
  }
```

Reply:
not sure if this needs error handling just a redirect to main page

Reply:
Yes correct.